tests/apps/test_similarity.py

```diff
@@ -368,71 +451,230 @@ def test_mean_event(self, monkeypatch):
 
             assert np.allclose(similarity.prob_event_mc(graph, 6, 3), 1.0)
 
-    def test_loss(self, monkeypatch):
-        """Test if function correctly creates the SF program for lossy GBS."""
-        graph = nx.complete_graph(5)
-        mock_eng_run = mock.MagicMock()
+    def test_max_loss(self):
+        """Test if function samples from the vacuum when maximum loss is applied."""
+        graph = nx.complete_graph(6)
+        assert similarity.prob_event_mc(graph, 6, 3, samples=1, loss=1) == 0.0
 
-        with monkeypatch.context() as m:
-            m.setattr(sf.LocalEngine, "run", mock_eng_run)
-            similarity.prob_event_mc(graph, 6, 3, samples=1, loss=0.5)
-            p_func = mock_eng_run.call_args[0][0]
 
-        assert isinstance(p_func.circuit[1].op, sf.ops.LossChannel)
+class TestFeatureVectorOrbits:
```

Comment:
Also needs to test that correct values are calculated for non-trivial specific cases